### PR TITLE
gui/kivy: fix installwizard back button behaviour

### PR DESCRIPTION
**Pull request by zebra-lucky**
fix android back button behavior in gui/kivy/uix/dialogs/installwizard.py

As in https://github.com/akhavr/electrum-dash/issues/88

### CODE REVIEW

File: electrum/gui/kivy/uix/dialogs/installwizard.py [hunk 1]
```diff
@@ -562,6 +565,20 @@ def add_widget(self, widget, index=0):
         else:
             self.crcontent.add_widget(widget, index=index)
 
+    def on_keyboard(self, instance, key, keycode, codepoint, modifier):
+        if key == 27:
+            if self.can_go_back:
```
Comment (SomberNight) [inline electrum/gui/kivy/uix/dialogs/installwizard.py]: `can_go_back` is a method, so this will always be True

Comment (SomberNight) [reply electrum/gui/kivy/uix/dialogs/installwizard.py]: ahh no, disregard. there are two of them :)
`self.wizard.can_go_back()` and `self.can_go_back`
is `self.can_go_back` needed?

Comment (zebra-lucky) [reply electrum/gui/kivy/uix/dialogs/installwizard.py]: Yes, maybe I should use wizard method.
I'll post changes in a 5 minutes.

**Comment by zebra-lucky** [comment]
Removed can_go_back, use wizard method.
Rebased and squashed changes.

File: electrum/gui/kivy/uix/dialogs/installwizard.py (hunk 1 above)
Comment (zebra-lucky) [reply electrum/gui/kivy/uix/dialogs/installwizard.py]: Removed can_go_back, use wizard method.

**Comment by ecdsa** [comment]
how can the initial issue be reproduced?

**Comment by zebra-lucky** [comment]
On Android fresh install exit from installwizard on back button.
On Linux same behaviour can be tested with `Esc` button.

**Comment by ecdsa** [comment]
@SomberNight could you reproduce the issue?

**Comment by zebra-lucky** [comment]
Yes, I reproduced the problem on Android and Linux on 3.3.8.

**Comment by zebra-lucky** [comment]
Not sure about freezing, however.
Only exit from installwizard was reproduced and fixed.

**Comment by ecdsa** [comment]
@zebra-lucky can you post the traceback?

**Comment by zebra-lucky** [comment]
No, I not reproduced any tracebacks.
Only clean exit from app on this event from popup:

```python
    def on_dismiss(self):
        app = App.get_running_app()
        if app.wallet is None and not self._on_release:
            app.stop()
```

**Comment by ecdsa** [comment]
oh I see. I was not understanding the issue. yes, I can reproduce.

**Comment by SomberNight** [comment]
Callbacks bound to Window need to be unbound too. See https://github.com/spesmilo/electrum/commit/95ae42b998f6abbe8667d9032f2084f78ad009bf
This was missing even before this PR but it was harmless then.
Without fix, pressing back button after creating a new wallet would go back to the wizard (after it was supposedly done).

**Comment by zebra-lucky** [comment]
Thanks, I missed unbinding when doing patch.

**Comment by zebra-lucky** [comment]
Found another bug. Need to call `self.close()` before go back. As in:
```python
        if key == 27:
            if self.wizard.can_go_back():
                self.close()
                self.wizard.go_back()
```

And there is a problem with `App.on_stop`.
If in pause mode and app window is closed, then `on_stop` method is not called.
Maybe should call `save_last_wallet` from `on_pause`
```
        if self.wallet:
            self.electrum_config.save_last_wallet(self.wallet)
```
And maybe also should call `network.stop` from `on_pause` and `network.start` from `on_resume`

Will do some PR in couple of hours.

**Pull request merged**